.github/actions/on_device_tests/action.yaml

```diff
@@ -40,6 +40,14 @@ runs:
           echo "TEST_TYPE=unit_test" >> $GITHUB_ENV
           echo "USE_SHARDING=1" >> $GITHUB_ENV
         fi
+
+        # Mobile Harness platform override.
+        if [ "${{ matrix.mh_platform }}" != "" ]; then
+          echo "MH_PLATFORM=${{ matrix.mh_platform }}" >> $GITHUB_ENV
+        else
+          # Default to target_platform.
```

Comment:
I'm not sure it should be doing this--this would lead to more implicit issues, where we're sending the wrong platform to MH instead of failing earlier when we can't retrieve matrix.mh_platform

Reply:
Are you suggesting we explicitly set 'mh_platform' parameter everywhere and stop relying on 'target_platform' altogether?

Reply:
Yes

Reply:
done

Reply:
changed parameter to 'device_test_platform'